### ISSUE TITLE
Drag folder from finder to warp dock icon and have it open

### ISSUE THREAD

**Issue by elviskahoro** [warpdotdev/Warp#992]
### Describe the solution you'd like?

```text
Drag folder from finder to warp dock icon and have it open
```

**Comment by mnhattt** [comment]
+1

**Comment by elviskahoro** [comment]
No workarounds unfortunately

**Comment by elviskahoro** [comment]
This has been launched!

**Comment by nickroberts** [comment]
Just tried it. It works, however, Warp opens 2 windows. 1, the directory I dragged onto the icon, and another, the default (last used?!?) directory.

Getting close! Now, just need some background transparency, and launch settings!

**Comment by nickroberts** [comment]
I tried turning off the `Restore windows, tabs, and panels on startup` feature, cleared the sqlite db, and when dragging a folder to the application icon, it still opens up a new window with my home directory, along with the folder I dropped onto the icon.

**Comment by elviskahoro** [comment]
Can you share a video? It just opens up a new tab with that directory for me

**Comment by nickroberts** [comment]
Ok, I think I figured out the issue. It seems as if I misinterpreted what the functionality was supposed to do. I just too a look at the changelog, and saw this:

```
Drag and drop a folder or file onto the Warp dock icon to open a new tab in this directory
```

Which, if I have Warp already open, and drag a folder to the Warp icon, it does just that, it opens a new tab in the directory.

What I was expecting, was if I didn't have Warp open at all, and I dragged a folder onto the application icon, it would open Warp in that directory. 1 Warp window, with 1 tab, the directory I dropped onto the app icon.

It will open 2 Warp windows, 1 with my home directory, the other with the directory I dropped.

If I close all of the open Warp windows, but keep it running (don't `Quit Warp` or hit `⌘ Q`), then drop a folder on the app icon, it opens a new window, with a single tab with the dropped directory.

I guess it might be a bit nit-picky on my end, but that's how I have always used iTerm, and it works great.
